scripts/build_scripts/build_with_profiler_opt.sh

```diff
@@ -11,7 +11,7 @@ if [[ -z "$ARCH_NAME" ]]; then
     exit 1
 fi
 
-cmake -B build -G Ninja -DENABLE_TRACY=ON
```

Comment:
Could you make building tests an arg option. Tests builds almost double the build times and in most scenarios they are not needed so we should default on not building them.

Reply:
@mo-tenstorrent I am struggling to understand why this `build_with_profiler_opt.sh` script even exists.

Do you use it?

Most of the workflows directly call CMake, but this one workflow depends on this script.

We also have another script that can build with the profiler, and there I am adding optional support for building with tests...

https://github.com/tenstorrent/tt-metal/pull/13259/files#r1781704195

Reply:
If you want changes in `build_with_profiler_opt.sh` lets discuss it in a new issue / pr ... there might be some cleanup we can do there.

Reply:
This script precedes  `build_metal.sh` and our move to CMAKE. YAMLs used to call it from day one and it stayed that after moving to CMAKE as well.

The `-p` option is a very recent add to `build_metal.sh` and in its commit this script should have been retired tbh

It does a clean non-incremental profiler build which I am assuming `-p` on `build_metal.sh` does as well now. 

I am all for retiring it now.  On CI YAMLs, we can call CMAKE directly as well.

Reply:
Thanks for the background, I'll log an issue.

Reply:
No problem. How about, whatever test that requires this, we call cmake in them directly instead and remove the call to this script from them.